nixos/modules/services/networking/wireguard.nix

```diff
@@ -246,12 +246,16 @@ let
         };
 
         script = ''
-          mkdir --mode 0644 -p "${dirOf values.privateKeyFile}"
+          set -e
+
+          # If the parent dir does not already exist, create it.
+          # Otherwise, does nothing, keeping existing permisions intact.
+          mkdir -p --mode 0755 "${dirOf values.privateKeyFile}"
+
           if [ ! -f "${values.privateKeyFile}" ]; then
-            touch "${values.privateKeyFile}"
-            chmod 0600 "${values.privateKeyFile}"
+            # Ensure private key file cannot be read by other users ahead of creation.
+            (set -e; umask 077; touch "${values.privateKeyFile}")
             wg genkey > "${values.privateKeyFile}"
```

Comment:
Why not
```suggestion
            (set -e; umask 077; wg genkey > "${values.privateKeyFile}")
```
?

Reply:
That's also possible and closer to what e.g. Archwiki does. I hadn't done it to stay closer to what it is now, but you're right, might as well do it directly.

I'll change it.

Reply:
Done.